src/training/adversarial_training.jl

```diff
@@ -11,12 +11,10 @@ function vanilla_train(
     batch_size;
     loss = logitcrossentropy,
     opt = Adam,
-    min_label = 0,
-    max_label = 9,
 )
     θ = Flux.params(model)
     vanilla_losses = []
-    train_loader = DataLoader((x_train, y_train), batchsize = batch_size, shuffle = true)
+    train_loader = DataLoader((x_train, y_train), batchsize = batch_size, shuffle = true) |> gpu
```

Comment:
Should this be `Flux.get_device()` to keep it general? https://fluxml.ai/Flux.jl/stable/guide/gpu/#Using-device-objects

Reply:
Neat, didn't know that was a thing. I'll change this, and I'll also add some tests for the train, and the attacks (simple ones for now to check if the perturbation is within bounds)